src/wasm/wasm-validator.cpp

```diff
@@ -3457,6 +3462,13 @@ void FunctionValidator::visitFunction(Function* curr) {
   }
 }
 
+void FunctionValidator::validateOffset(Address offset,
+                                       Memory* mem,
+                                       Expression* curr) {
+  shouldBeTrue(
+    mem->is64() || offset < (1ull << 32), curr, "offset must be u32");
```

Comment:
```suggestion
    mem->is64() || offset <= std::limits<uint32_t>::max(), curr, "offset must be u32");
```

Reply:
We use that pattern elsewhere. But if it leads to annoying issues with compiler warnings on signed comparisons, feel free to ignore.